HERACLES-33: Historical & Lab Results range/unit revisions

sashaandjic opened the pull request:
https://phenotips.atlassian.net/browse/HERACLES-33

sashaandjic committed:
HERACLES-33: Historical & Lab Results range/unit revisions

Updated Laboratory Results questionnaire

acrowthe commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Laboratory Results.xml:
```diff
@@ -530,14 +618,27 @@
+			<value>15.0</value>
+			<type>Double</type>
+		</property>
+		<property>
```
Two instances of `<property>`, remove one of them

sashaandjic commented:
@Hajany here is the look of the modified Laboratory Results form with min, max and ranges being added:
![Laboratoty Results](https://github.com/user-attachments/assets/6f05f8c6-ba93-44d9-95b9-c1dc08b59121)

If the ranges are making it look too busy I can removed them.

marta- commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Laboratory Results.xml:
```diff
@@ -562,14 +661,29 @@
 		<property>
 			<name>unitOfMeasurement</name>
-			<value>cells/mcL</value>
+			<value>thousand/&eta;L</value>
```
`µ` is `mu`, not `eta`. There's no need to escape it, as it doesn't break the XML syntax. Also, this named entity is not recognized in XML, so it's not rendered.
```suggestion
			<value>thousand/µL</value>
```

marta- commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Laboratory Results.xml:
```diff
@@ -591,14 +705,30 @@
 		<property>
 			<name>unitOfMeasurement</name>
-			<value>cells/mcL</value>
+			<value>million/&eta;L</value>
```
```suggestion
			<value>million/µL</value>
```

marta- commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Laboratory Results.xml:
```diff
@@ -562,14 +661,29 @@
 		<property>
 			<name>unitOfMeasurement</name>
-			<value>cells/mcL</value>
+			<value>thousand/&mu;L</value>
```
There's no need to escape it, as it doesn't break the XML syntax. Also, this named entity is not recognized in XML, so it's not rendered. Note how in the UI this unit of measurement appears as "thousand/L". Use the character `µ` instead.
```suggestion
			<value>thousand/µL</value>
```

marta- commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Laboratory Results.xml:
```diff
@@ -591,14 +705,30 @@
 		<property>
 			<name>unitOfMeasurement</name>
-			<value>cells/mcL</value>
+			<value>million/&mu;L</value>
```
```suggestion
			<value>million/µL</value>
```

sashaandjic replied:
Thank you. I already pushed the changes.

sashaandjic commented:
Here is the new screenshot of the Lab results form with µL in units for white and red blood cells.
![Laboratoty Results](https://github.com/user-attachments/assets/6c602ede-b3db-4727-a292-ed276047b879)

sashaandjic committed:
HERACLES-33: Historical & Lab Results range/unit revisions

Updated Historical Laboratory Results questionnaire
* added min and max values
* added value ranges in the description of the question
* added units of measurement to pro_BNP question

sashaandjic commented:
Screenshot of the Historical Lab results form:
![Historical Laboratoty Results](https://github.com/user-attachments/assets/354c4138-8007-4fd9-b978-3f9c4f3e0f55)

marta- commented:
Rebased on the latest `dev`, squashed commits for cleaner history.

sdumitriu committed:
HERACLES-33: Historical & Lab Results range/unit revisions
Bugfix: <  and > aren't displayed

sdumitriu commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Laboratory Results.xml:
```diff
@@ -678,14 +840,29 @@
 		</property>
+		<property>
+			<name>description</name>
+			<value>Range: 80-100</value>
```
The description says 80-100, but the allowed range is 70-120. Is this OK? Maybe call this "Expected range" or "Normal range"?

Hajany replied:
This is ok, warning should only appear for any value lower than 70 or higher than 120

sdumitriu commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Laboratory Results.xml:
```diff
@@ -707,11 +884,26 @@
 		</property>
+		<property>
+			<name>description</name>
+			<value>Range: 0 - 3</value>
```
This range is even weirder, the maximum allowed is 200, way above the expected 3.

sashaandjic replied:
@Hajany I noticed this too, but did not adjust it as I though maybe the expected range is for the healthy individuals, but we allow a bit more for "unhealthy" ones.

Hajany replied:
Confirmed that this range is expected. The expected range is a reference value for "normal" but CRP can be very high as 200 in some cases. Warning should only be displayed if value is lower than 0 or higher than 200

sdumitriu reviewed (approved):
I fixed the `<` and `>` not showing up. Are they sure that no value will ever be out of range? Because a value out of range makes the form invalid, which means it will not be exported.

sashaandjic committed:
HERACLES-33: Historical & Lab Results range/unit revisions

Restored spaces that were removed in the previous commit from markdown-formatted descriptions

Hajany commented:
> @Hajany here is the look of the modified Laboratory Results form with min, max and ranges being added: ![Laboratoty Results](https://private-user-images.githubusercontent.com/23243244/364610036-6f05f8c6-ba93-44d9-95b9-c1dc08b59121.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1ODI0NjIsIm5iZiI6MTcyNjU4MjE2MiwicGF0aCI6Ii8yMzI0MzI0NC8zNjQ2MTAwMzYtNmYwNWY4YzYtYmE5My00NGQ5LTk1YjktYzFkYzA4YjU5MTIxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE3VDE0MDkyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBmMjVhYTQwMzIzNTExZTk5NmMzNTc4ODU1MGI2NjAwNWJkNDgyOTVlYWIxYjg3YTljZGVmNDBkYmM5NDg1MzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AaLJlvP63Qdj19CpaJbi5-h92DiH58C7qQKbrVQcwe4)
> 
> If the ranges are making it look too busy I can remove them.

Leave the ranges as descriptions, study team wants them to visible in the database as a reference

sashaandjic commented:
![Laboratoty Results latest](https://github.com/user-attachments/assets/0483ef5c-3cdd-40cd-b961-5147b4650c6a)
![Historical Laboratoty Results latest](https://github.com/user-attachments/assets/59f090cf-c155-4b71-ac4f-8e753692124c)

sdumitriu commented:
Hemoglobin A1c seems to be missing newlines in both forms. Is that intentional?

sashaandjic commented:
> Hemoglobin A1c seems to be missing newlines in both forms. Is that intentional?

no. i missed that one. and red blood cells too.

sashaandjic reviewed (commented):
all spaces seem to be back now

sashaandjic commented:
2 more bugs:
Historical Lab results:
![image](https://github.com/user-attachments/assets/c6dd0bbf-2a82-4d9d-a738-d4a5224c8659)
Lab results:
![image](https://github.com/user-attachments/assets/1f0a8662-baf7-49fc-9416-21a18cfa58d5)

sashaandjic committed:
HERACLES-33: Historical & Lab Results range/unit revisions

* Corrected unit of measurements for Lab Results CRP
* Added unit of measurements to Hisotrical Lab Results CRP
* Corrected pro-BNP range in Historical Lab Results

sashaandjic commented:
tested. all seems fine.

marta- committed:
HERACLES-33: Historical & Lab Results range/unit revisions

Disabled the enforcement of min and max values for lab results

marta- commented:
* Squashed the commits that restored trailing spaces in markdown descriptions
* Re-labeled all commits that did not have the issue name and summary in their commit message
* Rebased on #1810 
* Updated the two questionnaires to display a custom message for values out of range, per specs

sashaandjic commented:
tested. looks good.
[Historical LAB.pdf](https://github.com/user-attachments/files/17050381/Historical.LAB.pdf)
[LAB.pdf](https://github.com/user-attachments/files/17050383/LAB.pdf)

sashaandjic commented:
@marta - one small thing, When I enabled the reinforcement for Cholesterol, the error message was displayed in 1 line.
![image](https://github.com/user-attachments/assets/b566e4d7-7074-4652-a757-ee4b2f9dc815)

marta- commented:
> When I enabled the reinforcement for Cholesterol, the error message was displayed in 1 line. 
> Also, we no longer have ability to show min/max values.

Updated branch `CARDS-2580` to display the default message ("Please enter values between...") instead of the custom one when min/max are enforced. Rebased this branch for testing.

sashaandjic commented:
TESTED - looks good

One very minor thing that we can address later. If not enforcing the min/max, display the message bellow the input line.
I really like the new message being displayed under the input line, it would improve the look if we always do it.
![image](https://github.com/user-attachments/assets/65ce404c-92f6-47e4-8087-2ee8f7c7fa0a)

Hajany commented:
Tested - minor changes:
![image](https://github.com/user-attachments/assets/ec740c3b-a6a0-4937-8b7e-ec1a7dc6f905)
![image](https://github.com/user-attachments/assets/d7980dbf-c9c7-40c2-9040-423eeaa8aeab)
![image](https://github.com/user-attachments/assets/9fde4ad5-380c-4a31-b755-6f683efea925)

sashaandjic committed:
HERACLES-33: Historical & Lab Results range/unit revisions
Minor edits to question descriptions

sashaandjic commented:
> Tested - minor changes: ![image](https://private-user-images.githubusercontent.com/96542282/369265286-ec740c3b-a6a0-4937-8b7e-ec1a7dc6f905.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY4MTMzMjQsIm5iZiI6MTcyNjgxMzAyNCwicGF0aCI6Ii85NjU0MjI4Mi8zNjkyNjUyODYtZWM3NDBjM2ItYTZhMC00OTM3LThiN2UtZWMxYTdkYzZmOTA1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIwVDA2MTcwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc1M2YyOWVjZTI1NGVjNWE0MTIwMGFmZDc3MzkyYmJmMTdiOWEzYjk4NmU2NTdkNWY3Nzk1YmE2OTYyYjFmYTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.l_5ShEav14lGIpmqN7cXl93LYoc9ZHj7DoGYwJ3DqZo) ![image](https://private-user-images.githubusercontent.com/96542282/369265323-d7980dbf-c9c7-40c2-9040-423eeaa8aeab.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY4MTMzMjQsIm5iZiI6MTcyNjgxMzAyNCwicGF0aCI6Ii85NjU0MjI4Mi8zNjkyNjUzMjMtZDc5ODBkYmYtYzljNy00MGMyLTkwNDAtNDIzZWVhYThhZWFiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIwVDA2MTcwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM4OTY1MTU2Nzg4MGMxNzRlYTFlYjljYzYwMzIwOGJjNTE1YWM0NDBmNjcwM2I1OTUwN2IzNjEzOTM2NjY5NzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DMZNu-nZqAfxSDg4nFvux-Y2m5jeuTrPVWluzr1WJL4) ![image](https://private-user-images.githubusercontent.com/96542282/369265415-9fde4ad5-380c-4a31-b755-6f683efea925.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY4MTMzMjQsIm5iZiI6MTcyNjgxMzAyNCwicGF0aCI6Ii85NjU0MjI4Mi8zNjkyNjU0MTUtOWZkZTRhZDUtMzgwYy00YTMxLWI3NTUtNmY2ODNlZmVhOTI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIwVDA2MTcwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY0ZDMzMDVhY2M1ODNhNjQ1MTQ1MmVkMTZkYzM5MzYxM2EyMjMzZDI5Y2ViMDQ2M2M1ZGE0NjM3MzU2MTk4NjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8V-D0Ow3CUJF_5f7_VKBYu_A8akJgr6wGn5LW3pHaXs)

Addressed

marta- commented:
@sdumitriu Hold the merge until #1810 is approved and merged.

acrowthe reviewed (approved):
Looks good, tested

sashaandjic committed:
HERACLES-33: Historical & Lab Results range/unit revisions
Changed “Total Cholesterol” field name to Cholesterol
Changed “Fasting Blood Glucose” field name to Glucose

Hajany reviewed (approved):
tested
